### PR TITLE
Add reroute command for shard allocation

### PR DESCRIPTION
**Pull request by szabolcsf**
- Add new 'reroute' command to support manual shard allocation
- Support for allocating replica shards (safe operation)
- Support for allocating empty primary shards with data loss confirmation
- Interactive confirmation dialog for dangerous primary shard operations
- Additional options: --explain and --retry-failed
- Proper ATD type definitions for reroute operations
- JSON serialization and HTTP request handling
- Comprehensive help documentation with examples

Usage:
  es reroute <cluster> -r <index>:<shard>:<node>  # allocate replica es reroute <cluster> -p <index>:<shard>:<node>  # allocate empty primary

The command follows the same patterns as existing elasticsearch-cli commands and integrates seamlessly with the existing codebase.

### CODE REVIEW

File: src/es.ml [hunk 1]
```diff
@@ -1964,6 +2033,66 @@ let refresh_tool =
...
+let reroute_tool =
+  reroute_tool,
+  let open Term in
+  let doc = "reroute shards (allocate replica or empty primary shards)" in
+  let exits = default_exits in
+  let man = [
+    `S Manpage.s_description;
+    `P "The reroute command allows manual allocation of shards to specific nodes.";
+    `P "Use -r to allocate replica shards (safe operation).";
+    `P "Use -p to allocate empty primary shards (WARNING: causes data loss!).";
```
Comment (ahrrv) [inline src/es.ml:2087]: I think these should not be needed as it just repeats the args' documentation.

File: src/es.ml [hunk 2]
```diff
@@ -1964,6 +2033,66 @@ let refresh_tool =
...
+    Arg.(value & opt_all string [] & info [ "r"; "allocate-replica"; ] ~docv:"INDEX:SHARD:NODE" ~doc)
+  and allocate_empty_primary =
+    let doc = "allocate empty primary shard to node (format: INDEX:SHARD:NODE) - WARNING: CAUSES DATA LOSS!" in
+    Arg.(value & opt_all string [] & info [ "p"; "allocate-empty-primary"; ] ~docv:"INDEX:SHARD:NODE" ~doc)
+  and explain = Arg.(value & flag & info [ "e"; "explain"; ] ~doc:"explain the reroute decisions")
+  and retry_failed = Arg.(value & flag & info [ "f"; "retry-failed"; ] ~doc:"retry failed allocations")
+  in
+  let parse_allocation spec =
+    match String.split_on_char ':' spec with
+    | [index; shard_str; node] ->
```
Comment (ahrrv) [inline src/es.ml:2051]: for this kind of options, need to create a converter using Arg.conv (I think there are also a few ready-made converters under Arg.Conv).

File: src/es.ml [hunk 3]
```diff
@@ -1964,6 +2033,66 @@ let refresh_tool =
...
+      (match int_of_string shard_str with
+       | shard -> Some (index, shard, node)
+       | exception _ -> None)
+    | _ -> None
+  in
+  let replica_actions = 
+    List.filter_map (fun spec ->
```
Comment (ahrrv) [inline src/es.ml:2058]: filter_map and the only valid return is Some ...
LLM is drunk

File: src/es.ml [hunk 4]
```diff
@@ -1068,6 +1068,75 @@ let refresh { verbose; _ } {
...
+        Lwt.return (Some cmd)
+      | AllocateEmptyPrimary { index; shard; node; } ->
+        let%lwt confirmed = confirm_data_loss index shard node in
+        if confirmed then
+          let cmd = { Elastic_t.allocate_replica = None;
+                     allocate_empty_primary = Some { index; shard; node; accept_data_loss = true; };
+                     move = None; cancel = None; } in
+          Lwt.return (Some cmd)
+        else
+          let%lwt () = Lwt_io.eprintl "Operation cancelled." in
```
Comment (ahrrv) [inline src/es.ml:1117]: I think this should fail the whole operation. That is, all-or-nothing semantics.

File: src/elastic.atd [hunk 5]
```diff
@@ -234,3 +234,47 @@ type bulk_action = {
   inherit doc_id;
   ?routing : string option;
 }
+
+type reroute_command = {
+  ?allocate_replica : allocate_replica option;
+  ?allocate_empty_primary : allocate_empty_primary option;
+  ?move : move_shard option;
+  ?cancel : cancel_shard option;
```
Comment (ahrrv) [inline src/elastic.atd:242]: These are in ATD but not available as commands.
